presto-spi/src/main/java/com/facebook/presto/spi/function/JavaAggregationFunctionImplementation.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.function;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.function.aggregation.Accumulator;
+import com.facebook.presto.spi.function.aggregation.AggregationMetadata;
+import com.facebook.presto.spi.function.aggregation.GroupedAccumulator;
+
+import java.util.List;
+
+public interface JavaAggregationFunctionImplementation
+        extends AggregationFunctionImplementation
+{
+    boolean isDecomposable();
+
+    List<Type> getParameterTypes();
+
+    Type getIntermediateType();
+
+    Type getFinalType();
+
+    boolean isOrderSensitive();
+
+    default List<Class> getLambdaInterfaces()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    default AggregationMetadata getAggregationMetadata()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    default Class<? extends Accumulator> getAccumulatorClass()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    default Class<? extends GroupedAccumulator> getGroupedAccumulatorClass()
+    {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
I don't think you need to have default implementation for any of these functions.

Reply:
OK